tests/test_node.py

```diff
@@ -46,50 +43,61 @@ def Count(self):
             self.ord_list.append(3)
             return self
 
-    def test_dfs_order_small(self):
-        t = DfsTest.Temp()
+    @classmethod
+    def traverse(cls, node, prev_object= None):
+        """
+        Do a depth-first traversal of the graph from the root
 
-        node = Node(None, None)
-        node.value = t
+        """
+        if not node.operation:
+            prev_object = DfsTest.Temp()
+            node.value = prev_object
+            node._graph_prune()
 
-        node._dfs(node = node._get_head())
+        else:
+            ## Execution of the node
+            op = getattr(prev_object, node.operation.name)
+            node.value = op(*node.operation.args, **node.operation.kwargs)
 
-        self.assertEqual(t.ord_list, [])
+        for n in node.next_nodes:
+            DfsTest.traverse(n, node.value)
 
-    def test_dfs_order_big(self):
-        t = DfsTest.Temp()
+        return prev_object.ord_list
+
+
+    def test_dfs_graph_with_pruning(self):
```

Comment:
So the question here was: what are `test_dfs_graph_with_pruning` and `test_dfs_graph_without_pruning` testing in `test_node.py` that is not being tested already by `test_mapper_without_pruning` and `test_mapper_with_pruning` in `test_callable_generator.py`?

Reply:
@etejedor , this function (`test_dfs_graph_with_pruning`) must be renamed to `test_graph_pruning`. And `test_dfs_graph_without_pruning` will have to be renamed to `test_no_pruning`. My mistake that I forgot to rename these. It's main purpose is to check if the nodes are pruned correctly and this uses a dfs function that I defined in the current test class itself to check the order of nodes. `test_mapper_without_pruning` and `test_mapper_with_pruning` test the mapper function that gets generated with and without graph pruning. This already assumes that graph pruning is correct and doesn't test for pruning at all

Reply:
Isn't this line:
https://github.com/shravan97/PyTDF/blob/6d8ee4cd80a112ba9c7344ff2ebe563148da6108/tests/test_callable_generator.py#L57
precisely testing the pruning in `test_callable_generator.py`?

And this line:
https://github.com/shravan97/PyTDF/blob/6d8ee4cd80a112ba9c7344ff2ebe563148da6108/tests/test_node.py#L83
again testing the pruning in `test_node.py`?

Reply:
@etejedor , the function `test_mapper_with_pruning` is just a test case to make sure that the mapper function works as it is even when some parts of the graph are pruned. It makes sure that the mapper function responds to the changes the graph even after pruning. I'm not very sure if this is compulsory though. I'm also getting a feeling of a slight redundancy here.

Only difference is that, the first test (`test_callable_generator.py`) assumes that the pruning function is correct and checks for generated mapper and the second one (`test_node.py`) assumes that the dfs function (`traverse()`) is correct and only tests for correct pruning.

Reply:
How can `test_callable_generator.py` assume that the pruning function is correct, when there is this check?
https://github.com/shravan97/PyTDF/blob/6d8ee4cd80a112ba9c7344ff2ebe563148da6108/tests/test_callable_generator.py#L59

Don't you end up calling `graph_prune` in the same way you do it here:
https://github.com/shravan97/PyTDF/blob/6d8ee4cd80a112ba9c7344ff2ebe563148da6108/tests/test_node.py#L55

Reply:
@etejedor , the first check (in `test_callable_generator.py`) is to verify that the mapper traversed the graph in the order that we want despite pruning the graph. In this case `graph_prune` is called here : https://github.com/shravan97/PyTDF/blob/4f93732f059555f123495f3c5ed3eae4d6601f05/PyTDF/CallableGenerator.py#L67